packages/apidom-ns-workflows-1/src/refractor/visitors/workflows-1/SourceDescriptionsVisitor.ts

```diff
@@ -0,0 +1,31 @@
+import stampit from 'stampit';
+import { ArrayElement, Element, BREAK } from '@swagger-api/apidom-core';
+
+import SourceDescriptionsElement from '../../../elements/nces/SourceDescriptions';
+import SpecificationVisitor from '../SpecificationVisitor';
+import FallbackVisitor from '../FallbackVisitor';
+import { isSourceDescriptionLikeElement } from '../../predicates';
+
+const SourceDescriptionsVisitor = stampit(SpecificationVisitor, FallbackVisitor, {
+  init() {
+    this.element = new SourceDescriptionsElement();
+  },
+  methods: {
+    ArrayElement(arrayElement: ArrayElement) {
+      arrayElement.forEach((item: Element) => {
```

Comment:
We don't need to assert the shape here. The look can look like this:

```ts
      arrayElement.forEach((item: Element): void => {
        const specPath = ['document', 'objects', 'SourceDescription'];
        const element = this.toRefractedElement(specPath, item);

        this.element.push(element);
      });
```

The reason is that the [SourceDescriptionVisitor](https://github.com/swagger-api/apidom/pull/3403/files#diff-f3530852a6bd15c65b79b08c2a0a74408bad87c3e8ac52c8acd48a65dbdc9538R8) contains `FallbackVisitor` in it's prototype chain and will either refract to semantic `SourceDescription` or generic value.